src/Lock.php

```diff
@@ -526,6 +526,176 @@ public static function showForItem(CommonDBTM $item)
+                    'itemtype'   => $item->getType(),
+                    'is_dynamic' => 1,
+                    'is_deleted' => 1
+                ]
+            );
+        }
+
+        //networkport
+        $iterator = $DB->request([
+            'SELECT'    => ['id'],
+            'FROM'      => NetworkPort::getTable(),
+            'WHERE'     => [
+                'is_dynamic'   => 1,
+                'is_deleted'   => 1,
+                'items_id'     => $item->fields['id'],
+                'itemtype'     => $item->getType(),
+            ]
+        ]);
+        $nb += $iterator->count();
+
+        //count networkname
+        //see Lock.php:381
+        foreach ($iterator as $networPortID) {
+            $iterator = $DB->request([
+                'SELECT' => 'id',
+                'FROM'   => NetworkName::getTable(),
+                'WHERE'  => [
+                    'itemtype'     => 'NetworkPort',
+                    'items_id'     => $networPortID,
+                    'is_deleted'   => 1,
+                    'is_dynamic'   => 1
+                ]
+            ]);
+            $nb += $iterator->count();
+
+            //count ipaddress
+            //see Lock.php:412
+            foreach ($iterator as $networNameID) {
+                $iterator = $DB->request([
+                    'SELECT' => 'id',
+                    'FROM'   => IPAddress::getTable(),
+                    'WHERE'  => [
+                        'itemtype'     => 'NetworkName',
+                        'items_id'     => $networNameID,
+                        'mainitems_id' => $item->fields['id'],
+                        'mainitemtype' => $item->getType(),
+                        'is_deleted'   => 1,
+                        'is_dynamic'   => 1
+                    ]
+                ]);
+                $nb += $iterator->count();
+            }
+        }
+
+        //Software versions
+        $iterator = $DB->request([
+            'SELECT'    => [
+                'isv.id AS id',
+                'sv.name AS version',
+                's.name AS software'
+            ],
+            'FROM'      => Item_SoftwareVersion::getTable() . " AS isv",
+            'LEFT JOIN' => [
+                'glpi_softwareversions AS sv' => [
+                    'FKEY' => [
+                        'isv' => 'softwareversions_id',
+                        'sv'  => 'id'
+                    ]
+                ],
+                'glpi_softwares AS s'         => [
+                    'FKEY' => [
+                        'sv'  => 'softwares_id',
+                        's'   => 'id'
+                    ]
+                ]
+            ],
+            'WHERE'     => [
+                'isv.is_deleted'  => 1,
+                'isv.is_dynamic'  => 1,
+                'isv.items_id'     => $item->fields['id'],
+                'isv.itemtype'     => $item->getType(),
+            ]
+        ]);
+        $nb += $iterator->count();
```

Comment:
On this one, I do not get why there are joined tables, as there is no WHERE condition on them.

Reply:
done !